kartograf/rpki/fetch.py

```diff
@@ -106,26 +105,29 @@ def process_file(file):
                                  "-P",
                                  context.epoch,
                                  ] + tal_options +
-                                 ["-f", file],  # -f has to be last
+                                 ["-f"] + batch,  # -f has to be last
                                  capture_output=True,
                                  check=False)
 
         if result.stderr and context.debug_log:
             stderr_output = result.stderr.decode()
             with debug_file_lock:
                 with open(context.debug_log, 'a') as logs:
-                    logs.write(f'\nfile: {file}\n')
                     logs.write(stderr_output)
         return result.stdout
 
-    with ThreadPoolExecutor() as executor:
-        results = list(tqdm(executor.map(process_file, files), total=len(files)))
-
-    json_results = [result.decode() for result in results if result]
+    total = len(files)
+    batch_size = 250
 
     with open(result_path, "w") as res_file:
         res_file.write("[")
-        res_file.write(",".join(json_results))
-        res_file.write("]")
+        for i in tqdm(range(0, total, batch_size)):
+            batch = [str(f) for f in files[i:i + batch_size]]
+            results = process_file(batch)
+            normalized = results.replace(b"\n}\n{\n\t", b"\n},\n{\n").decode('utf-8').strip()
+            res_file.write(normalized)
+            res_file.write(",")
+
+        res_file.write("{}]")
```

Comment:
this is a dumb hack, but I couldn't think of a way to not write a final comma delimiter? i.e. know that its the last thing to process. This results in having at least one `incomplete` counted in the parsing results.